python/hsfs/expectation_suite.py

```diff
@@ -128,37 +159,267 @@ def to_ge_type(self):
             meta=self._meta,
         )
 
-    def __str__(self):
+    def _init_feature_store_and_feature_group_ids_from_href(self, href: str) -> None:
+        self._feature_store_id, self._feature_group_id = re.search(
+            r"\/featurestores\/([0-9]+)\/featuregroups\/([0-9]+)\/expectationsuite*",
+            href,
+        ).groups(0)
+
+    def _init_expectation_engine(
+        self, feature_store_id: int, feature_group_id: int
+    ) -> None:
+        if self.id:
+            self._expectation_engine = ExpectationEngine(
+                feature_store_id=feature_store_id,
+                feature_group_id=feature_group_id,
+                expectation_suite_id=self.id,
+            )
+        else:
+            raise ValueError(
+                "Initialise the Expectation Suite first by attaching to a Feature Group"
+            )
+
+    def _init_expectation_suite_engine(
+        self, feature_store_id: Optional[int], feature_group_id: Optional[int]
+    ) -> None:
+        if feature_group_id and feature_store_id:
+            self._expectation_suite_engine = (
+                expectation_suite_engine.ExpectationSuiteEngine(
+                    feature_store_id=feature_store_id,
+                    feature_group_id=feature_group_id,
+                )
+            )
+        elif self._feature_group_id and self._feature_store_id:
+            self._expectation_suite_engine = (
+                expectation_suite_engine.ExpectationSuiteEngine(
+                    feature_store_id=self._feature_store_id,
+                    feature_group_id=self._feature_group_id,
+                )
+            )
+        else:
+            raise ValueError(
+                "Provide feature_store_id or feature_group_id to use the expectation suite API"
+            )
+
+    # Expectation Suite Convenience Operations
+    def save(
```

Comment:
These kind of methods we usually keep in the "engine" classes, e.g. `ExpectationSuiteEngine`

Reply:
Yes, I am a bit worried people use `expectation_suite.expectations.append(my_expectation)` and then want to persist it. I will remove it